### PR TITLE
Validation test

### PR DESCRIPTION
**Pull request by vanessuniq**
# Summary

This PR adds validation test to all profile groups. The validation test verifies resources returned from the read and search tests conform to the given profile. It does so by using the external validator service to verify the presence of mandatory elements and that elements with required bindings contain appropriate values.
The test will be skipped if no resource read or search requests were made in previous tests or if all requests were unsuccessful. 
The test will fail if at least one of the resources returned does not conform to the given profile. 

This PR also adds presets to the PFE suite.

# Testing

- Review the code logic and ensure it makes sense.
- Run unit tests and ensure passing
- Run the test in the browser  and ensure correct behavior.

### CODE REVIEW

File: spec/pacio_test_kit/validation_test_spec.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    allow_any_instance_of(runnable).to receive(:tag).and_return(profile)
+  end
+
+  it 'passes when at least one previous read or search request is successful and returns conformant resource' do
+    mock_server(body: observation, valid_resource: true)
+    result = run(runnable)
+    expect(result.result).to eq('pass')
+  end
+
+  it 'skips when no read or search requests were made in presvious tests' do
```
Comment (KaelynJefferson) [inline spec/pacio_test_kit/validation_test_spec.rb:62]: presvious --> previous

File: lib/pacio_test_kit/common_tests/validation_test.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    def tag
+      config.options[:profile]
+    end
+
+    def profile_url
+      PACIO_PROFILES[tag]
+    end
+
+    run do
```
Comment (KaelynJefferson) [inline lib/pacio_test_kit/common_tests/validation_test.rb:34]: So as the validation_test is now, we only tell the user that resources of a resource_type are successful/validated etc. We don't mention which request it is? Example: If we read several id's, we just let the user know that only some of them returned were successful, but the user won't know the specific id if there is an error?

File: spec/pacio_test_kit/validation_test_spec.rb (hunk 1 above)
Comment (vanessuniq) [reply spec/pacio_test_kit/validation_test_spec.rb:62]:
```suggestion
  it 'skips when no read or search requests were made in previous tests' do
```

File: lib/pacio_test_kit/common_tests/validation_test.rb (hunk 2 above)
Comment (vanessuniq) [reply lib/pacio_test_kit/common_tests/validation_test.rb:34]: The requests executed are displayed on the browser. The user will know which request failed. And if the validation fails for a resource, the validator will specify the resource with the given issue.

Screenshot 2024-07-14 at 3 48 35 PM

Screenshot 2024-07-14 at 3 50 16 PM

**Review by KaelynJefferson — APPROVED** [review]
Looks good: reviewed code, ran unit tests, browser app ran as expected.

**Pull request merged**